From a8f4a2032f69ef31d0269ec763e2df8d9abd5239 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@freescale.com>
Date: Thu, 26 Nov 2015 17:34:18 +0800
Subject: [PATCH 0682/5242] MLK-11896-2 usb: chipidea: imx: derefence pointer
 after NULL check

commit  3843becbac49568557aeda2b2113822817e9930d from
https://source.codeaurora.org/external/imx/linux-imx.git

Dereference data pointer after its NULL check.

Signed-off-by: Li Jun <jun.li@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/chipidea/usbmisc_imx.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/chipidea/usbmisc_imx.c b/drivers/usb/chipidea/usbmisc_imx.c
index 5f0375c..a43d390 100644
--- a/drivers/usb/chipidea/usbmisc_imx.c
+++ b/drivers/usb/chipidea/usbmisc_imx.c
@@ -1073,12 +1073,13 @@ int imx_usbmisc_set_wakeup(struct imx_usbmisc_data *data, bool enabled)
 int imx_usbmisc_charger_detection(struct imx_usbmisc_data *data, bool connect)
 {
 	struct imx_usbmisc *usbmisc;
-	struct usb_charger *charger = data->charger;
+	struct usb_charger *charger;
 	int ret = 0;
 
 	if (!data)
 		return -EINVAL;
 
+	charger = data->charger;
 	usbmisc = dev_get_drvdata(data->dev);
 	if (!usbmisc->ops->charger_primary_detection)
 		return -ENOTSUPP;
-- 
1.7.9.5

